### PR TITLE
Purge 3.6.0 / Disable the late runtime processor when using CLI 

### PR DESCRIPTION
**Pull request by kasperg**
#### Link to issue

https://reload.atlassian.net/browse/DDFNEXT-359
https://reload.atlassian.net/browse/DDFNEXT-364

#### Description

When running drush deploy on production environments we are frequently
seeing problems where our process fails with the following error:

Drupal\Core\Database\DatabaseExceptionWrapper: SQLSTATE[HY000]:
General error: 2006 MySQL server has gone away: SELECT * FROM
"purge_queue" q WHERE ((expire = 0) OR (:now > expire)) ORDER BY
created, item_id ASC LIMIT 0, 100;

It occurs for little over 10% of production deployments. Redeploying
the site will make the problem go away.

This query occurs because the late runtime processor tries to claim
items from the purge queue at the end of Drush commands.

We have not been able to identify the root cause of this issue. Based
on the metrics and logs we have available the database does not seem
to be under significant strain. There should not be a large quantity
of items in the queue etc.

To avoid this from happening we have discussed alternate solutions:

- Reducing the number of items claimed does not make a difference. We
tried going from 100 to 30.
- We could move the purge queue to alternate storage. Here Redis comes
to mind since it is already used in the project. However it is worth
noting that this would require Redis with persistence to avoid loosing
items if Redis restarts or is moved within the cluster. Our current
use of Redis is not persistent and this would also require introducing
a sandbox module to support Redis with Purge.
https://www.drupal.org/sandbox/marvin_b8/3417595

In the end we go with a simpler solution: Not implicitly processing
the purge queue when running CLI commands. This way we also avoid
it during deployment Drush commands which is where this error occurs.

For deployments this should not make a difference. We already
explictly purge the entire Varnish Cache when doing a deployment.
https://github.com/danskernesdigitalebibliotek/dpl-platform/blob/main/infrastructure/dpladm/env-repo-template/standard/.lagoon.yml#L31

There might be other situations where an update which happens during
cron invoked with drush cli would place items in the purge queue. Here
these items would not be processed immediately. They will have to wait
for the next event which will trigger purge queue processing e.g.
an uncached web request or a drush cron.

We expect both events to occur frequently enough that this should not
be a problem.

To support this change we first update the Purge module from 3.5 to 3.6. It is something that we needed to do anyway.

### CODE REVIEW

**Comment by rasben** [comment]
It looks good to me, but, I prefer that it's @spaceo that reviews and takes 'ownership' of this, if it is to be merged whilst you are gone, Kasper :)

**Review by spaceo — APPROVED** [review]
LGTM 👍 
I have some comments for you to look at

File: cypress/e2e/varnish.cy.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  title: "Varnish test",
+  subtitle: "A subtitle",
+  path: "/articles/varnish-test",
+};
+
+const varnishCacheHeader = "x-varnish-cache";
+
+describe("Varnish", () => {
+  it("is caching responses for anonymous users", () => {
+    cy.session("anonymous", () => {});
```
[the remark this answers is not in the document]
Comment (kasperg) [reply cypress/e2e/varnish.cy.ts]: This is an outdated commit. The final change uses `anonymousUser()` like what is referenced in the suggestion.

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -78,14 +78,18 @@ Cypress.Commands.add("logRequests", () => {
...
+  const password = Cypress.env("DRUPAL_PASSWORD");
+  cy.session({ username, password }, () => {
+    cy.visit("/user/login");
+    cy.get('[name="name"]')
+      .type(username)
+      .parent()
+      .get('[name="pass"]')
+      .type(password);
+    cy.get('[value="Log in"]').click();
+  });
```
[the remark this answers is not in the document]
Comment (kasperg) [reply cypress/support/commands.ts:91]: It is not entirely out of context. For this test I wanted to switch between an editorial and an anonymous user during the test. I have good experience doing so using `cy.session()` from `cypress/e2e/patron-collisions.cy.ts` so I wanted to use the same approach here.

I have tried to explain the reason behind this in the commit commit.

**Pull request merged**